### ISSUE TITLE
💅 noSecrets false positives: URLs with query strings, ANSI escapes, and camelCase identifiers

### ISSUE THREAD

**Issue by maks-oleksyuk** [biomejs/biome#11323]
### Environment information

<details>

```bash
CLI:
  Version:                      2.5.8
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           linux

Environment:
  BIOME_DISTRIBUTION:                npm
  BIOME_LOG_PATH:                    unset
  BIOME_LOG_PREFIX_NAME:             unset
  BIOME_LOG_LEVEL:                   unset
  BIOME_LOG_KIND:                    unset
  BIOME_CONFIG_PATH:                 unset
  BIOME_THREADS:                     unset
  BIOME_WATCHER_KIND:                unset
  BIOME_WATCHER_POLLING_INTERVAL:    unset
  NO_COLOR:                     unset
  TERM:                         xterm
  JS_RUNTIME_VERSION:           v26.7.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Not set
  Path:                         unset

Workspace:
  Open Documents:               0
```

</details>

### Rule name

`security/noSecrets`

### Playground link

https://biomejs.dev/playground/?lintRules=noSecrets&tab=formatter&pane=Diagnostics&code=YwBvAG4AcwB0ACAAdQByAGwAIAA9ACAAJwBoAHQAdABwADoALwAvAGwAbwBjAGEAbABoAG8AcwB0AC8AYQBwAGkALwBkAHUAbQBtAHkAagBzAG8AbgAvAHAAcgBvAGQAdQBjAHQAcwA%2FAGwAaQBtAGkAdAA9ADIAJwA7AAoAYwBvAG4AcwB0ACAAYQBuAHMAaQAgAD0AIAAnAFwAeAAxAGIAWwAzADEAbQBoAGkAXAB4ADEAYgBbADAAbQAnADsACgBjAG8AbgBzAHQAIAByAGUAbAAgAD0AIAAnAEcARQBUACAALwBzAGUAYQByAGMAaAA%2FAHEAPQAlADIAMAB0AGUAcwB0ACcAOwAKAGMAbwBuAHMAdAAgAHIAZQBsADIAIAA9ACAAJwAvAHAAbwBzAHQAcwA%2FAHAAYQBnAGUAPQAxACYAcQA9AGEAKwBiACcAOwAKAGMAbwBuAHMAdAAgAHQAaQB0AGwAZQAgAD0AIAAnAGcAZQB0AEQAdQBtAG0AeQBKAHMAbwBuAEIAYQBzAGUAVQByAGwAJwA7AA%3D%3D

All 5 lines are flagged as "Potential secret found." under default config (`entropyThreshold: 41`):

```ts
// 1. URL with query string — the safe-pattern whitelist's URL regex
// (^https?://[a-zA-Z0-9.-]+(/[a-zA-Z0-9./_-]*)?$) doesn't allow `?`/`=`/`&`,
// so any URL with query params falls straight through to entropy scoring.
const url = 'http://localhost/api/dummyjson/products?limit=2';

// 2. ANSI escape sequence
const ansi = '\x1b[31mhi\x1b[0m';

// 3. HTTP method + relative path + query string (no scheme, so the URL
// whitelist never applies to begin with)
const rel = 'GET /search?q=%20test';

// 4. Relative URL with query string
const rel2 = '/posts?page=1&q=a+b';

// 5. Plain camelCase identifier (same class as the Tailwind-class/CDK-ID
// false positives in #9114 / #8136 / #4113 — frequent case transitions
// inflate the entropy score without the string having real entropy)
const title = 'getDummyJsonBaseUrl';
```

### Expected result

None of these five should be flagged — they're an ordinary URL, an ANSI color code, two relative paths, and a camelCase identifier.

For case 1 specifically, the root cause is fixable and narrow: the whitelist's "standard URL" regex needs an optional query-string group, e.g.:

```
^https?://[a-zA-Z0-9.-]+(/[a-zA-Z0-9./_-]*)?(\?[a-zA-Z0-9=&%_.+-]*)?$
```

Cases 2–5 fall under the broader entropy-formula false-positive class already tracked in #9114, #8136, and #4113 (mixed-case/symbol/digit strings scoring above threshold despite carrying no real secret entropy) — same weakness, different trigger strings.

**Comment by ematipico** [comment]
We will deprecate the rule. Use other tools that fit the job
